### PR TITLE
tests: fifo_timeout : Dereference after null check

### LINKED ISSUE #8063: [Coverity CID :186190] Null pointer dereferences in /tests/kernel/fifo/fifo_timeout/src/main.c
Static code scan issues seen in File: /tests/kernel/fifo/fifo_timeout/src/main.c
 Category: Null pointer dereferences
 Function: test_multiple_threads_get_data
 Component: Tests
 CID: 186190
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by hegderaj4**
fifo data obtained is dereferenced after NULL check.

Coverity-CID:186190
Fixes #8063

Signed-off-by: Nagaraj Hegde <hegdenagaraj4@gmail.com>

### CODE REVIEW

**Comment by hegdenagaraj4** [comment]
I had to upload the changes again due to rebase issue.

File: tests/kernel/fifo/fifo_timeout/src/main.c
Comment (hegdenagaraj4) [inline tests/kernel/fifo/fifo_timeout/src/main.c]: Changed data->timeout_order to data->q_order.

File: tests/kernel/fifo/fifo_timeout/src/main.c
Comment (andyross) [inline tests/kernel/fifo/fifo_timeout/src/main.c]: Wouldn't it have been simpler just to put "if(!data) { return TC_FAIL; }" right here instead of mucking with the nested if structure?

**Review by andyross — APPROVED** [review]
Seems more complicated than it should be, and the commit message is a little obfuscatory (How about "Need to check return value of k_fifo_get() for NULL"), but no reason not to commit.

File: tests/kernel/fifo/fifo_timeout/src/main.c
Comment (hegdenagaraj4) [inline tests/kernel/fifo/fifo_timeout/src/main.c]: <if statement > -> if the data is NULL then, the TC_ERROR is dereferencing data->q_order without NULL check. It will crash. Hence the change.

File: tests/kernel/fifo/fifo_timeout/src/main.c
Comment (hegdenagaraj4) [reply tests/kernel/fifo/fifo_timeout/src/main.c]: In this if(!data) { return TC_FAIL; } then TC_ERROR print should be deleted. Hence did not do this change. Thought that print should be necessary.

Comment (hegdenagaraj4) [reply tests/kernel/fifo/fifo_timeout/src/main.c]: Hi @andyross , I know you approved it, but this is not merged. Would like to know if I need to make changes you mentioned to merge this. I am new, trying to understand how merge happens usually?

Comment (andyross) [reply tests/kernel/fifo/fifo_timeout/src/main.c]: Anas generally picks these up and merges them once all the checks have passed and approval is done.  I'm not demanding changes here, it was just a nitpick.

Comment (hegdenagaraj4) [reply tests/kernel/fifo/fifo_timeout/src/main.c]: Cool thanks 👍 will wait.

**Review by nashif — APPROVED** [review]

**Pull request merged**
